File: spark-java/spark/src/main/java/demo/utils/spk/MySparkUtil.java
package demo.utils.spk;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaSparkContext;

/**
 * @author admin 2020-7-13
 */
public class MySparkUtil {

    public static JavaSparkContext jsc(String master, String appName) {
        SparkConf conf = new SparkConf().setAppName(appName).setMaster(master);
        JavaSparkContext jsc = new JavaSparkContext(conf);
        jsc.setLogLevel("WARN");
        return jsc;
    }

    public static JavaSparkContext jsc(String appName) {
        SparkConf conf = new SparkConf().setAppName(appName).setMaster("local");
        JavaSparkContext jsc = new JavaSparkContext(conf);
        jsc.setLogLevel("WARN");
        return jsc;
    }

}

File: spark-scala-mvn/src/main/scala/utils/MySparkUtil.scala
package utils

import org.apache.spark.sql.SparkSession
import org.apache.spark.{SparkConf, SparkContext}

/**
 *
 * @author admin 2020-5-26
 */
object MySparkUtil {
  val conf = new SparkConf()

  def sc(master: String, appName: String) = {
    conf.setMaster(master).setAppName(appName)
    new SparkContext(conf)
  }

  def sc(appName: String) = {
    conf.setMaster("local").setAppName(appName)
    val sparkContext = new SparkContext(conf)
    sparkContext.setLogLevel("WARN")
    sparkContext
  }

  def spark(master: String, appName: String) = {
    SparkSession.builder().master(master).appName(appName).getOrCreate()
  }

}
